### ISSUE TITLE
[0.9.6.1] TSM crash - out of memory

### ISSUE THREAD

**Issue by miili** [influxdata/influxdb#5270]
InfluxDB running with UDP and local Telegraf input surprisingly crashed after running stable for 3 weeks.
Series created and memory usage went up due to no obvious reason nor change of the configuration.

Further the data after 8:00 is lost now and does not show up in Grafana anymore. See crashlog and influxdb internals

![20160102_influx_crash](https://cloud.githubusercontent.com/assets/4992805/12113635/358f6398-b3a4-11e5-8adf-2a41b08f089b.jpg)

https://gist.github.com/miili/bf38d84e03a6c1d982a1

**Comment by kenzodeluxe** [comment]
Waiting for this to happen, too:
```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                                
13359 root      20   0 60.636g 0.024t 273320 S   4.6 77.4 342:01.15 influxd 
```

**Comment by beckettsean** [comment]
> Series created and memory usage went up due to no obvious reason

If the series cardinality rises, so do the RAM requirements. 

Why did the number of new series climb? What is the approximate total series cardinality? Are you writing from any sources other than telegraf? Did you update any of the telegraf installations or configurations?

**Comment by miili** [comment]
We write through UDP, a single, small and stable embedded agent, besides Telegraf writes locally for local monitoring.

The configuration did not change for at least 14d prior to the crash. Therefore I am surpised to see the series created sky rocket.

Unfortunately I could not find any telefgraf logs on the system, neither any hints in syslog.

**Comment by beckettsean** [comment]
Telegraf logs to `/var/log/telegraf/telegraf.log` by default.

@miili is the data still missing or does it appear in Grafana now? If it is still missing from Grafana, is it directly queryable from the CLI or curl?

Has the series count continued to rise? Is the system otherwise healthy or does it OOM on restart?

**Comment by jwilder** [comment]
@miili This looks like a compaction ran and tried to allocate memory, but failed.  There are a number of fixes since 0.9.6.1 that might have fixed this.  Is this still occurring with current nightlies?

**Comment by toddboom** [comment]
Closing this out because I believe it's fixed in the v0.10.0 release of TSM. Please let us know if you still see it with the latest version and we'll reopen.
